### PR TITLE
Fix package for latest wagmi, ethers and rainbow-kit versions & integrate Magic Connect

### PR DESCRIPTION
**Pull request by tchataigner**
Hey everyone, Thomas from the [Polyphene](https://polyphene.io/) team here ! 

With @PhilippeMts and myself stumbled upon this package while looking to integrate [Magic Connect](https://magic.link/docs/connect/overview) with Rainbow Kit on our latest product [Crowd Connect](https://crowdconnect.xyz/).

As some development were required to make it work with the latest versions of Ethers and Rainbow Kit here is a PR to propose changes to integrate them.

> Note: As the Magic Connect API varies quite a bit from Magic Auth we took the liberty of exporting 2 connectors from the crate. This should be sanitize in the future when the Connect API gets  better.

### Major

- Creation of two classes `MagicAuthConnector` & `MagicConnectConnector`
  - `MagicAuthConnector`: Connector integrating with [Magic Auth](https://magic.link/docs/auth/overview). Most of the code comes from previous implementation
  - `MagicConnectConnector`: Connector integrating with [Magic Connect](https://magic.link/docs/connect/overview).
- Made `MagicConnector` an abstract class containing shared logic between `MagicAuthConnector` & `MagicConnectConnector`
- Renamed `options.additionalMagicOptions` to `options.magicSdkConfiguration`, which seemed to be a clearer name
- Updated documentation in README to fit changes

### Minor

- Fixed some typos in the README
- Fixed Rainbow Kit example in the README and specified that `options.magicSdkConfiguration.network.chainId` needs to be specified. This comes from the fact that in their most recent version Rainbow Kit makes a `getChainId()` call on the connector before calling the `connect()` method.
- Fixed typo in enableSMSlogin -> enableSMSLogin

## Next steps

Once this PR has been iterated on and merged, we will also update the example repository.

Right now the `autoConnect` property of wagmi on `MagicConnectConnector` is ensured through a value stored in local storage. This has to be done because calling a method of the Magic Connect API opens the Magic modal and the `Magic.user` methods are not available for Magic Connect. 
This would have to be changed when Magic connect API gets better.

## Related issues 

1. https://github.com/EveripediaNetwork/wagmi-magic-connector/issues/32

This PR should fix compatibility problem related to newest version of Rainbow Kit. @brightiron feel free to take a look at this iteration.

2. https://github.com/EveripediaNetwork/wagmi-magic-connector/issues/17

This PR makes the usage of Magic Connect possible. @thomashug feel free to take a look at this iteration.

Hope everything seems clear. I'd be glad to discuss/iterate on any

### CODE REVIEW

**Comment by tchataigner** [comment]
Last commit meant for `changeset`, I was not used to this automation 😅

**Comment by Royal-lobster** [comment]
@tchataigner Thanks for opening the PR, This is really awesome. we will review this and merge asap.

**Comment by tchataigner** [comment]
Happy to collaborate ! Feel free to ping me on any unclear part or any needed iteration !!

**Comment by brightiron** [comment]
This is great. I was debugging this yesterday and came to the same conclusion around getChainId() being called first, and not opening the modal if options.magicSdkConfiguration.network is not set. 

One potential improvement for the future would to just use the same rpc/network that's already being passed by rainbow/wagmi so it doesn't have to be configured again.

But loving the improvements!!

File: src/lib/connectors/magicConnector.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    enableEmailLogin: boolean,
+    oauthProviders: OAuthProvider[]
+  ): Promise<UserDetails> {
+    const output: UserDetails = (await createModal({
+      accentColor: this.magicOptions.accentColor,
+      isDarkMode: this.magicOptions.isDarkMode,
+      customLogo: this.magicOptions.customLogo,
+      customHeaderText: this.magicOptions.customHeaderText,
+      enableSMSLogin: enableSMSLogin,
+      enableEmailLogin: enableEmailLogin,
```
Comment (Royal-lobster) [inline src/lib/connectors/magicConnector.ts:66]:
```suggestion
      enableEmailLogin: enableEmailLogin || true,
```
This makes the email login to be enabled by default.

**Review by Royal-lobster — CHANGES_REQUESTED** [review]

**Comment by Royal-lobster** [comment]
Great ! everything looks good for the merge.

**Comment by tchataigner** [comment]
Waow so fast ! That's really nice, thank you very much 🙏

**Comment by Royal-lobster** [comment]
> Waow so fast ! That's really nice, thank you very much 🙏

Thanks for the contribution 😇

**Pull request merged**